Fix BOM parsing in RFC5424

stevedlawrence opened the pull request:
The message part of an RFC5424 syslog message can start with a BOM to signify UTF-8, or no BOM to signify ASCII. This was implemented using a BOM initiator, but it was specified incorrectly. Instead of specifying three separate hex bytes that make up a UTF-8 BOM (EF BB EF), we really needed to specify the Unicode code point (FEFF) and that is encoded into the correct bytes for the the intiiator.

This wasn't detected because the TDML tests also specified the BOM using the same incorret DFDL entities. To avoid this kind of error, DFDL entities are no longer used in the TDML tests, and are instead specified using "byte" documentParts, making the bytes more explicit.

mbeckerle commented on src/main/resources/com/owlcyberdefense/syslog/xsd/syslog-rfc5424.dfdl.xsd:
```diff
@@ -204,7 +204,7 @@ SOFTWARE.
         <xs:complexType>
           <xs:choice>
-            <xs:element name="UTF8" dfdl:initiator="%#xEF;%#xBB;%#xBF;" type="sl:msgStr" dfdl:encoding="UTF-8" />
+            <xs:element name="UTF8" dfdl:initiator="%#xFEFF;" type="sl:msgStr" dfdl:encoding="UTF-8" />
```
I suggest adding annotation-documentation to this UTF8 element explaining that this is a BOM character (0xFEFF) which will be the 3 bytes EF BB BF in UTF-8 encoding.

mbeckerle reviewed (approved):
+1 
Minor suggestion to add a doc string.

stevedlawrence committed:
Fix BOM parsing in RFC5424

The message part of an RFC5424 syslog message can start with a BOM to
signify UTF-8, or no BOM to signify ASCII. This was implemented using a
BOM initiator, but it was specified incorrectly. Instead of specifying
three separate hex bytes that make up a UTF-8 BOM (EF BB EF), we really
needed to specify the Unicode code point (FEFF) and that is encoded into
the correct bytes for the the intiiator.

This wasn't detected because the TDML tests also specified the BOM using
the same incorret DFDL entities. To avoid this kind of error, DFDL
entities are no longer used in the TDML tests, and are instead specified
using "byte" documentParts, making the bytes more explicit.
